lib/rules/no-unused-prop-types.js

```diff
@@ -725,6 +725,70 @@ module.exports = {
     }
 
     /**
+     * Marks all props found inside ObjectTypeAnnotaiton as declared.
+     * 
+     * Modifies the declaredProperties object
+     * @param {ASTNode} propTypes 
+     * @param {Object} declaredPropTypes 
+     * @returns {Boolean} True if propTypes should be ignored (e.g. when a type can't be resolved, when it is imported)
+     */
+    function declarePropTypesForObjectTypeAnnotation(propTypes, declaredPropTypes) {
+      let ignorePropsValidation = false;
+
+      iterateProperties(propTypes.properties, (key, value) => {
+        if (!value) {
+          ignorePropsValidation = true;
+          return;
+        }
+
+        let types = buildTypeAnnotationDeclarationTypes(value, key);
+        if (types === true) {
```

Comment:
instead of reusing `types` here, can we use multiple vars and use `const`?

Reply:
This is from the existing code so didn't touch it yet. But I think we can simply do: 

```js
const types = buildTypeAnnotationDeclarationTypes(value, key) || {};
declaredPropTypes.push({
  ...types,
  fullName: key,
  name: key,
  node: value
});
```

Reply:
Meh, another problem is that `buildTypeAnnotationDeclarationTypes` returns either a boolean or an object which isn't great either. I try to refactor it to always return an object, but it needs changes in more places than just here. 

But I do think it's a good refactor to do 😄